threads/src/main/java/org/jboss/as/threads/ThreadFactoryAdd.java

```diff
@@ -55,10 +55,12 @@ private ThreadFactoryAdd() {
     protected void performRuntime(final OperationContext context, final ModelNode operation, final ModelNode model) throws OperationFailedException {
 
         ModelNode priorityModelNode = PoolAttributeDefinitions.PRIORITY.resolveModelAttribute(context, model);
+        ModelNode groupNameModelNode = PoolAttributeDefinitions.GROUP_NAME.resolveModelAttribute(context, model);
+        ModelNode threadNamePatternModelNode = PoolAttributeDefinitions.THREAD_NAME_PATTERN.resolveModelAttribute(context, model);
 
-        final String threadNamePattern = PoolAttributeDefinitions.THREAD_NAME_PATTERN.resolveModelAttribute(context, model).asString();
+        final String threadNamePattern = threadNamePatternModelNode.isDefined() ? threadNamePatternModelNode.asString() : null;
```

Comment:
This can all be simplified to 

```
final String threadNamePattern = 
        PoolAttributeDefinitions.THREAD_NAME_PATTERN.resolveModelAttribute(context, model).asStringOrNull();
final String groupName =
        PoolAttributeDefinitions.GROUP_NAME.resolveModelAttribute(context, model).asStringOrNull();
```

I added asXxxOrNull() methods to ModelNode some time this year.

Reply:
This can be dealt with some other time. The method already uses the old pattern for the other attribute anyway.